core/controllers/reader_test.py

```diff
@@ -159,6 +159,49 @@ def test_welcome_exploration(self):
             'Finnish', 'Yes! Oppia is the Finnish word for learn.',
             'What is the value of')
 
+    def test_soft_classification(self):
+        """Tests the string classifier exploration with responses that
+        trigger soft rules.
+        """
+        self.init_player(
+            '16', 'Testing String Classifier', 'do you think three things can')
+        self.submit_and_compare('Because you can arrange each colour '
```

Comment:
Nit (style): There should be a break after the '('. We break on function call parentheses before breaking up the string. E.g.:

```python
self.submit_and_compare(
    'Because you can arrange each colour '
    'systematically and there are two permutations of each colour',
    'Detected permutation', 'do you think three things can')
```

(though with the strings rearrange so they fit the entire line).

Please fix this here and elsewhere in this file.